n2y/errors.py

```diff
@@ -67,81 +61,93 @@ def __init__(self, response, message, code) -> None:
         self.code = code
 
 
-class ConnectionThrottled(HTTPResponseError):
+class ConnectionThrottled(APIResponseError):
     """
     Raised when the connection is throttled by the Notion API.
     """
 
     def __init__(self, response, message=None) -> None:
-        retry = response.headers.get("retry-after")
-        self.retry_after = float(retry) if retry else None
+        self.retry_after = (
+            float(retry) if (retry := response.headers.get("retry-after")) else 0
+        )
         if message is None:
             message = (
                 "Your connection has been throttled by the Notion API for"
                 f" {self.retry_after} seconds. Please try again later."
             )
-        super().__init__(response, message)
+        super().__init__(response, message, APIErrorCode.RateLimited)
 
 
 class ObjectNotFound(APIResponseError):
     def __init__(self, response, message) -> None:
-        code = APIErrorCode.ObjectNotFound
-        super().__init__(response, f"{message} [{code}]", code)
-        self.code = code
+        super().__init__(response, message, APIErrorCode.ObjectNotFound)
 
 
-class APIErrorCode(str, Enum):
-    Unauthorized = "unauthorized"
-    """The bearer token is not valid."""
+class APIErrorCode(StrEnum):
+    is_retryable: bool
 
-    RestrictedResource = "restricted_resource"
-    """Given the bearer token used, the client doesn't have permission to
-    perform this operation."""
+    def __new__(cls, code: str, is_retryable: bool):
+        obj = str.__new__(cls, code)
+        obj._value_ = code
+        obj.is_retryable = is_retryable
+        cls.RetryableCodes = [i.value for i in cls if i.is_retryable is True]
```

Comment:
I think `is True` here is unnecessary. And the next line can be `if not i.is_retryable`.

Reply:
True, lol